cryostatio/cryostat-operator
# feat(insights): optionally deploy a proxy for Insights

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Related to: https://github.com/cryostatio/cryostat/issues/1763

## Description of the change:
This change is expected to do nothing unless the environment variable `INSIGHTS_ENABLED` is set to true. If the variable is set to true, then the operator takes several actions:
1. Creates a secret containing an APICast configuration containing the `cloud.openshift.com` authentication token from the global pull secret.
2. Creates a corresponding APICast deployment and service that appends an authentication header using the credentials from the previous step, before forwarding requests to the Insights server.
3. Configures all Cryostat deployments to find the APICast proxy at the URL within the `INSIGHTS_PROXY` environment variable.

## Motivation for the change:
Red Hat customers will automatically be able to integrate Red Hat Insights with their workloads by using the Cryostat Operator, Cryostat, and Cryostat Agent, from the downstream Red Hat build of Cryostat.

## How to manually test:
1. `make oci-build SKIP_TESTS=true ENABLE_INSIGHTS=true OPERATOR_IMG=quay.io/ebaron/cryostat-operator:insights-test-11`
4. `make deploy ENABLE_INSIGHTS=true OPERATOR_IMG=quay.io/ebaron/cryostat-operator:insights-test-11`
5. Observe proxy deployment, service, and APICast config secret in cryostat-operator-system namespace
6. `make cryostat_cr`
7. Cryostat deployment should have `INSIGHTS_PROXY` set to `http://insights-proxy.cryostat-operator-system.svc.cluster.local`.
8. When deployed without `ENABLE_INSIGHTS=true`, none of the objects from step 3 should be present, and the `INSIGHTS_PROXY` environment variable should not be set.
